students/jesse_miller/session03/mailroom.py

```diff
@@ -0,0 +1,159 @@
+Beginning of my mailroom implementation.
+"""
+import sys
+
+DONORS = {"Robert Smith" : [435.56, 125.23, 357.10],
+          "JD Cronise" : [123.12],
+          "Chris Stapleton" : [243.87, 111.32],
+          "Dave Lombardo" : [63.23, 422.87, 9432.01],
+          "Randy Blythe" : [223.50, 8120.32],
+          "Devin Townsand" : [431.12, 342.92, 5412.45],
+          }
+print("")
+PROMPT = "\n".join(("Welcome to mailroom 0.1!",
+                    "",
+                    "Please choose from below options:",
+                    "mail - If you would like the mail menu.",
+                    "report - If you would like a report of donations totals.",
+                    "quit - Exit.",
+                    ">>> "))
+
+SEND_PROMPT = "\n".join(("Donor and Mail Database",
+                         "",
+                         "Please choose from below options:",
+                         "send - If you would like to send a thank you.",
+                         "list - If you would like to see a list of donors.",
+                         "back - If you would like to return to the main menu.",
+                         ">>> "))
+
+def mail_menu():
+    '''
+    This is the menu for the mail section
+    '''
+    while True:
+        mail_input = input(SEND_PROMPT)
+        if mail_input.lower() == 'list':
+            print("")
+            donor_list()
+            print("")
+        if mail_input.lower() == 'send':
+            print("")
+            donor_mail()
+        if mail_input.lower() == 'back':
+            print("")
+            main()
+        else:
+            print("Not a valid option")
+            print("")
+
+def report():
+    '''
+    This will be the donation report section
+    '''
+    summary= []
+    headers = ["Donor Name", "Total Given", "Times Donated", "Average Gift"]
+    print()
+    print("-"*80)
+    print("{:17} | {:<20} | {:<15} | {:<19}".format(headers[0], headers[1], headers[2], headers[3]))
+    print("-"*80)
+
+    for k,v in DONORS.items():
+        donor = k
+        total = (sum(v))
+        times = (len(v))
+        avg = (sum(v) / len(v))
+        summary.append([k, total, times, avg])
+    summary.sort(key=lambda d: d[1], reverse=True)
+    for x in summary:
+        print("{:17} |  ${:<18,.2f} | {:<15} |  ${:<17,.2f}".format(x[0], x[1], x[2], x[3]))
+    print("-"*80)
+    print("")
+    main()
+
+def goodbye():
+    '''
+    Gracefully exits
+    '''
+    print("Goodbye!")
+    sys.exit()
+
+def donor_list():
+    '''
+    This when done properly, will print the list of donor names
+    '''
+    print("-" * 15)
+    print("List of donors: ")
+    print("-" * 15)
+    for donor in DONORS:
+        print(donor)
+    print("")
+    #mail_menu()
+
+def donor_mail():
+    """
+    This section allows the user to mail a donor
+    """
+    while True:
+        donor_list()
+        donor_send = str(input("Who would you like to mail: "))
+        for donor in DONORS:
+            if donor_send == donor:
```

Comment:
If they don't enter the very first donor in the list, it would consider it a new donor.  I think the logic you want is
```
if donor_send in donors:
    mail_send()
else:.....
```

Reply:
and remove the for loop